airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/lag.sql

```diff
@@ -0,0 +1,41 @@
+{# lag end_at cdc ------------------------------------------------------------------------------     #}
+{% macro lag_end_at(cursor_field, primary_key, emitted_at) -%}
+  {{ adapter.dispatch('lag_end_at')(cursor_field, primary_key, emitted_at) }}
+{%- endmacro %}
+
+{% macro default__lag_end_at(cursor_field, primary_key, emitted_at) -%}
+    lag({{ cursor_field }}) over (
+        partition by {{ primary_key }}
+        order by {{ cursor_field }} is null asc, {{ cursor_field }} desc, {{ emitted_at }} desc
+    )
+{%- endmacro %}
+
+{% macro oracle__lag_end_at(cursor_field, primary_key, emitted_at) -%}
+    lag({{ cursor_field }}) over (
+        partition by {{ primary_key }}
+        order by {{ cursor_field }} desc, {{ emitted_at }} desc
+    )
+{%- endmacro %}
```

Comment:
@ChristopheDuong I'm struggling finishing this macro.

Reply:
> The main problem is because primary_key is transformed to something like => id, quote('_col'), cast(usd to {{ dbt_utils.string() }} and it's a string.

So, for dbt-oracle to work, since you told me it needs explicit jinja calls, you would have to translate it into:
`id, {{ quote('_col') }}, cast(usd to {{ dbt_utils.string() }}`, is that correct?

Reply:
It not work because I'm sending this to the macro this is render as `quote('_col')`

Reply:
and it should be rendered as `{{ quote('_col') }}` (even though it's in a jinja call already?

Thus: https://github.com/airbytehq/airbyte/pull/5562/files#r701068019

Reply:
I excluded this macros and change the function in the sql builder.